flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/state/ConfigMapView.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.operator.autoscaler.state;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.util.Preconditions;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+
+class ConfigMapView {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConfigMapView.class);
+
+    enum State {
+        /** ConfigMap is only stored locally, not created in Kubernetes yet. */
+        NEEDS_CREATE,
+        /** ConfigMap exists in Kubernetes but there are newer local changes. */
+        NEEDS_UPDATE,
+        /** ConfigMap view reflects the actual contents of Kubernetes ConfigMap. */
+        UP_TO_DATE
+    }
+
+    private State state;
+
+    private ConfigMap configMap;
+
+    private final Function<ConfigMap, Resource<ConfigMap>> resourceRetriever;
+
+    public ConfigMapView(
+            ConfigMap configMap, Function<ConfigMap, Resource<ConfigMap>> resourceRetriever) {
```

Comment:
```suggestion
            ConfigMap newConfigMap, Function<ConfigMap, Resource<ConfigMap>> resourceRetriever) {
```

How about rename the configMap to `newConfigMap` or `emptyConfigMap`? It's clear to let us know it's only used when `resourceRetriever.get()` is null.

Reply:
Renamed to `configMapSkeleton`.